prowler/providers/aws/services/iam/lib/policy.py

```diff
@@ -310,12 +325,20 @@ def is_condition_block_restrictive(
                         # if cross account is not allowed check for each condition block looking for accounts
                         # different than default
                         if not is_cross_account_allowed:
-                            # if there is an arn/account without the source account -> we do not consider it safe
+                            # if there is an arn/account without the source account or trusted accounts -> we do not consider it safe
                             # here by default we assume is true and look for false entries
                             for item in condition_statement[condition_operator][value]:
-                                if source_account not in item:
-                                    is_condition_key_restrictive = False
-                                    break
+                                if (
+                                    "aws:sourcevpc" != value
+                                    and "aws:sourcevpce" != value
+                                ):
```

Comment:
Why this? 

Why don't you split this function into several smaller ones? It is difficult to follow.

Reply:
I only added this if because these conditions does not contain an AWS Account ID, so I just check if they are present.